spec/i18n_spec.rb

```diff
@@ -16,9 +16,42 @@ def translate_expectation(binding, inspection)
+    it { expect(translate_expectation('foo', 'HasBinding')).to eq('<strong>foo</strong> debe estar declarado') }
+
+    it { expect(translate_expectation('foo.bar', 'HasMethod')).to eq('el método <strong>foo.bar</strong> debe estar declarado') }
+    it { expect(translate_expectation('foo.bar', 'HasIf')).to eq('<strong>foo.bar</strong> debe usar if') }
+
+    it { expect(translate_expectation('Intransitive:foo', 'Not:HasLambda')).to eq('<strong>foo</strong> no debe emplear expresiones lambda') }
+
+    it { expect(translate_expectation('foo', 'HasUsage:=baz')).to eq('<strong>foo</strong> debe utilizar <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'HasUsage:~baz')).to eq('<strong>foo</strong> debe delegar en algo parecido a <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'HasMethod:~baz')).to eq('<strong>foo</strong> debe declarar un método parecido a <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'HasUsage:*')).to eq('<strong>foo</strong> debe ser utilizado') }
+    it { expect(translate_expectation('foo', 'HasMethod:*')).to eq('el método <strong>foo</strong> debe estar declarado') }
+    it { expect(translate_expectation('foo', 'HasUsage:baz')).to eq('<strong>foo</strong> debe utilizar <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'HasUsage')).to eq('<strong>foo</strong> debe ser utilizado') }
+    it { expect(translate_expectation('foo', 'HasForall')).to eq('<strong>foo</strong> debe utilizar forall') }
+
+    it { expect(translate_expectation('foo', 'Not:HasUsage:=baz')).to eq('<strong>foo</strong> no debe utilizar <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'Not:HasUsage:~baz')).to eq('<strong>foo</strong> no debe delegar en algo parecido a <strong>baz</strong>') }
+    it { expect(translate_expectation('foo', 'Not:HasUsage:*')).to eq('<strong>foo</strong> no debe ser utilizado') }
+    it { expect(translate_expectation('foo', 'Not:HasUsage')).to eq('<strong>foo</strong> no debe ser utilizado') }
     it { expect(translate_expectation('foo', 'Not:HasUsage:baz')).to eq('<strong>foo</strong> no debe utilizar <strong>baz</strong>') }
     it { expect(translate_expectation('foo', 'Not:HasLambda')).to eq('<strong>foo</strong> no debe emplear expresiones lambda') }
+
+    it { expect(translate_expectation('', 'HasClass')).to eq('la solución debe declarar clases') }
+    it { expect(translate_expectation('', 'Not:HasMethod')).to eq('la solución no debe declarar métodos') }
+    it { expect(translate_expectation('', 'Not:HasClass')).to eq('la solución no debe declarar clases') }
+
+    it { expect(translate_expectation('foo', 'HasObject:^')).to eq('<strong>foo</strong> debe declarar objetos') }
+    it { expect(translate_expectation('', 'Not:HasClass')).to eq('la solución no debe declarar clases') }
```

Comment:
Finally, i dropped the `^` by changing semantics of matching. 

In the beginning, an inspection was true for a binding X if X matched the inspection, or any declaration contained by X matched it. Because of that, I needed to implement an operator that meant _match exactly X but not it sub-declarations to remove ambiguities. For example, the expression `A HasBinding` could mean: 

   * _A exists_, which is true for `class A {}`
   * _A declares a binding_, which was is true `class A { attribute b } 

However, this was error prone. 

Now, I have changed mulang 2 semantics to mean _the inspection match X  but not its subdeclarations_. Also, I have renamed all inspection to better reveal intention: 

```
x HasArity:n                        -> DeclaresArityN:x
x HasBinding                        -> Declares:x
x HasDirectRecursion                -> DeclaresRecursively:x
x HasTypeDeclaration                -> DeclaresTypeAlias:x
x HasTypeSignature                  -> DeclaresTypeSignature:x
x HasVariable                       -> DeclaresVariable:x
x HasComposition                    -> x UsesComposition
x HasComprehension                  -> x UsesComprehension
x HasConditional                    -> x UsesConditional
x HasForeach                        -> x UsesForeach
x HasGuards                         -> x UsesGuards
x HasIf                             -> x UsesGuards
x HasLambda                         -> x UsesLambda
x HasRepeat                         -> x UsesRepeat
x HasRepeatOf                       -> x UsesRepeat
x HasUsage:g                        -> x Uses:g
x HasWhile                          -> x HasWhile
```

This means new inspections are not backward compatible with previous versions of mulang. However, now mumukit-inspection delivers a compatibility layer which automatically translates them.